packages/babel-helpers/src/helpers/set.ts

```diff
@@ -1,22 +1,32 @@
 /* @minVersion 7.0.0-beta.0 */
 
-import superPropBase from "superPropBase";
-import defineProperty from "defineProperty";
+import defineProperty from "./defineProperty.ts";
+import superPropBase from "./superPropBase.ts";
 
-function set(target, property, value, receiver) {
+type Object = {
+  __proto__?: any;
+} & { [key: string]: unknown };
+
+function set(
+  target: Object,
+  property: string | symbol,
+  value: any,
+  receiver?: any,
+): boolean {
+  var setImpl = set;
   if (typeof Reflect !== "undefined" && Reflect.set) {
-    set = Reflect.set;
+    setImpl = Reflect.set;
   } else {
-    set = function set(target, property, value, receiver) {
+    setImpl = function set(target, property, value, receiver) {
       var base = superPropBase(target, property);
       var desc;
 
       if (base) {
         desc = Object.getOwnPropertyDescriptor(base, property);
-        if (desc.set) {
+        if (desc && desc.set) {
           desc.set.call(receiver, value);
           return true;
-        } else if (!desc.writable) {
+        } else if (!desc || !desc.writable) {
```

Comment:
You don't need to guard `desc` here. `desc` will always be defined, this is an implementation detail, just use `!`.